tests/templates/test_subroutines/test_prepselprep.py

```diff
@@ -220,16 +215,27 @@ def test_block_encoding(self, lcu, control, wire_order, dim):
             )
         ],
     )
-    def test_queuing_ops(self, lcu, control, results):
+    def test_queuing_ops(self, lcu, control, expected):
         """Test that qml.PrepSelPrep queues operations in the correct order."""
-        with qml.tape.QuantumTape() as tape:
-            qml.PrepSelPrep(lcu, control=control)
+        # Test that `compute_decomposition` queues the right ops
+        prepselprep = qml.PrepSelPrep(lcu, control=control)
+        with qml.queuing.AnnotatedQueue() as q0:
+            prepselprep.compute_decomposition(lcu, control)
+
+        # Test that `compute_decomposition` queues the right ops
+        with qml.queuing.AnnotatedQueue() as q1:
+            prepselprep.decomposition()
+
+        for op0, op1, exp_op in zip(q0.queue, q1.queue, expected, strict=True):
+            qml.assert_equal(op0, exp_op)
+            qml.assert_equal(op1, exp_op)
+
+        # Test that PrepSelPrep de-queues its input
+        with qml.queuing.AnnotatedQueue() as q0:
+            qml.apply(lcu)
+            prepselprep = qml.PrepSelPrep(lcu, control=control)
 
-        for idx, val in enumerate(tape.expand().operations):
-            assert val.name == results[idx].name
-            assert len(val.parameters) == len(results[idx].parameters)
-            for a, b in zip(val.parameters, results[idx].parameters):
-                assert (a == b).all()
+        assert len(q0.queue) == 1 and q0.queue[0] == prepselprep
```

Comment:
Updated queuing test that is more to our standards and also checks for de-queuing the input.